cordapp-example/clients/src/main/kotlin/com/example/server/MainController.kt

```diff
@@ -0,0 +1,122 @@
+    }
+
+
+    /**
+     * Displays all IOU states that exist in the node's vault.
+     */
+    @GetMapping(value = "ious", produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
+    fun getIOUs() : ResponseEntity<List<StateAndRef<IOUState>>> {
+        return ResponseEntity.ok(proxy.vaultQueryBy<IOUState>().states)
+    }
+
+
+    /**
+     * Initiates a flow to agree an IOU between two parties.
+     *
+     * Once the flow finishes it will have written the IOU to ledger. Both the lender and the borrower will be able to
+     * see it when calling /spring/api/ious on their respective nodes.
+     *
+     * This end-point takes a Party name parameter as part of the path. If the serving node can't find the other party
+     * in its network map cache, it will return an HTTP bad request.
+     *
+     * The flow is invoked asynchronously. It returns a future when the flow's call() method returns.
+     */
+
+    @PostMapping(value = "create-iou", produces = arrayOf("text/plain"), headers = arrayOf("Content-Type=application/x-www-form-urlencoded"))
+    fun createIOU(request: HttpServletRequest): ResponseEntity<String> {
+        val iouValue = request.getParameter("iouValue").toInt()
+        val partyName = request.getParameter("partyName")
+        val partyX500Name = CordaX500Name.parse(partyName)
+        if (iouValue <= 0 ) {
+            return ResponseEntity.badRequest().body("Query parameter 'iouValue' must be non-negative.\n")
+        }
+        if (partyX500Name == null) {
+            return ResponseEntity.badRequest().body("Query parameter 'partyName' missing or has wrong format.\n")
+        }
+        val otherParty = proxy.wellKnownPartyFromX500Name(partyX500Name) ?: return ResponseEntity.badRequest().body("Party named $partyName cannot be found.\n")
+
+        return try {
+            val signedTx = proxy.startTrackedFlow(::Initiator, iouValue, otherParty).returnValue.getOrThrow()
+            ResponseEntity.status(HttpStatus.CREATED).body("Transaction id ${signedTx.id} committed to ledger.\n")
+
+        } catch (ex: Throwable) {
+            logger.error(ex.message, ex)
+            ResponseEntity.badRequest().body(ex.message!!)
+        }
+    }
+
+    /**
+     * Displays all IOU states that are created by Party.
+     */
+    @GetMapping(value = "my-ious", produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
+    fun myious(): ResponseEntity<List<StateAndRef<IOUState>>>  {
+        val generalCriteria = QueryCriteria.VaultQueryCriteria(Vault.StateStatus.ALL)
+        val results = builder {
+            var partyType = IOUSchemaV1.PersistentIOU::lenderName.equal(proxy.nodeInfo().legalIdentities.first().name.toString())
+            val customCriteria = QueryCriteria.VaultCustomQueryCriteria(partyType)
+            val criteria = generalCriteria.and(customCriteria)
+            val results = proxy.vaultQueryBy<IOUState>(criteria).states
+            return ResponseEntity.ok(results)
```

Comment:
@EricMcEvoyR3 Can you move the return outside of the builder block? Realise this wasn't you, just seems like a good time for clean-up :)

Reply:
Having trouble here, node having trouble filtering by itself. 
`Caused by: net.corda.core.node.services.VaultQueryException: Parsing error: Unable to locate Attribute  with the the given name [lenderName] on this ManagedType [net.corda.core.schemas.PersistentState]`

Reply:
Yeah, someone added this endpoint to just get the IOUs created by the node itself. Maybe change the description to make it clearer? Not sure about that error - is the field there in `PersistentIOU`?

Reply:
Odd - I think you'll have to dig into this one unfortunately :(

Reply:
Resolved it by writing it the expected way : `proxy.vaultQueryBy<IOUState>().states.filter { it.state.data.lender.equals(proxy.nodeInfo().legalIdentities.first()) }` and behaves as expected